src/AppInstallerRepositoryCore/Rest/Schema/Json/SearchResponseDeserializer.cpp

```
@@ -0,0 +1,98 @@
+        constexpr std::string_view PackageFamilyName = "PackageFamilyName"sv;
+        constexpr std::string_view ProductCode = "ProductCode"sv;
+        constexpr std::string_view Versions = "Versions"sv;
+        constexpr std::string_view PackageVersion = "Version"sv;
+        constexpr std::string_view Channel = "Channel"sv;
+    }
+
+    std::optional<IRestClient::SearchResult> SearchResponseDeserializer::Deserialize(const web::json::value& searchResponseObject) const
+    {
+        // Make search result from json output.
+        if (searchResponseObject.is_null())
+        {
+            return {};
+        }
+
+        IRestClient::SearchResult result;
+        try
+        {
+            std::optional<std::reference_wrapper<const web::json::array>> dataArray = JsonHelper::GetRawJsonArrayFromJsonNode(searchResponseObject, JsonHelper::GetJsonKeyNameString(Data));
+            if (!dataArray.has_value() || dataArray.value().get().size() == 0)
+            {
+                AICLI_LOG(Repo, Verbose, << "No search results returned.");
+                return result;
+            }
+
+            for (auto& manifestItem : dataArray.value().get())
+            {
+                std::optional<std::string> packageId = JsonHelper::GetRawStringValueFromJsonNode(manifestItem, JsonHelper::GetJsonKeyNameString(PackageIdentifier));
+                std::optional<std::string> packageName = JsonHelper::GetRawStringValueFromJsonNode(manifestItem, JsonHelper::GetJsonKeyNameString(PackageName));
+                std::optional<std::string> publisher = JsonHelper::GetRawStringValueFromJsonNode(manifestItem, JsonHelper::GetJsonKeyNameString(Publisher));
+
+                if (!JsonHelper::IsValidNonEmptyStringValue(packageId) || !JsonHelper::IsValidNonEmptyStringValue(packageName) || !JsonHelper::IsValidNonEmptyStringValue(packageName))
+                {
+                    AICLI_LOG(Repo, Verbose, << "Skipping manifest item because of missing required package fields.");

```

We should fail if required values are missing.